backend/common/src/main/java/ai/verta/modeldb/common/futures/Future.java

```diff
@@ -37,17 +41,20 @@
   private final CompletionStage<T> stage;
 
   private Future(CompletionStage<T> stage) {
-    if (!DEEP_TRACING_ENABLED) {
-      formattedStack = null;
-    } else {
+    if (DEEP_TRACING_ENABLED
```

Comment:
Why not use `DEEP_TRACING_ENABLED` instead of adding another property?  I assume it is because `DEEP_TRACING_ENABLED` has broader effects?

Reply:
Yes, deep tracing also includes creating spans for every future execution.